modules/core/arrow-instances-core/src/main/kotlin/arrow/instances/try.kt

```diff
@@ -1,11 +1,56 @@
 package arrow.instances
 
 import arrow.Kind
-import arrow.core.*
+import arrow.core.Either
+import arrow.core.Eval
+import arrow.core.Failure
+import arrow.core.ForTry
+import arrow.core.Success
+import arrow.core.Try
+import arrow.core.TryOf
+import arrow.core.fix
+import arrow.core.identity
+import arrow.core.recoverWith
 import arrow.instance
-import arrow.typeclasses.*
+import arrow.typeclasses.Applicative
+import arrow.typeclasses.ApplicativeError
+import arrow.typeclasses.Eq
+import arrow.typeclasses.Foldable
+import arrow.typeclasses.Functor
+import arrow.typeclasses.Monad
+import arrow.typeclasses.MonadError
+import arrow.typeclasses.Monoid
+import arrow.typeclasses.Semigroup
+import arrow.typeclasses.Show
+import arrow.typeclasses.Traverse
 import arrow.instances.traverse as tryTraverse
 
+@instance(Try::class)
+interface TrySemigroupInstance<A> : Semigroup<Try<A>> {
+
+  fun SG(): Semigroup<A>
+
+  override fun Try<A>.combine(b: Try<A>): Try<A> =
+    when (this) {
```

Comment:
I'd move these functions to the datatype, reference them here instead.

Reply:
I couldn't find a way to do that given the dependency on SG and the semigroup interface. Is there a good example to follow given that core doesn't import typeclasses to avoid a circular dependency?